src/drivers/distance_sensor/tfmini/TFMINI.cpp

```diff
@@ -212,8 +407,34 @@ TFMINI::collect()
 		return -EAGAIN;
 	}
 
+	int8_t signal_quality;
+
+	switch (_hw_model) {
+	case TFMINI_MODEL::MODEL_TFMINI: // TFMINI (0 to 3000)
+		signal_quality = -1; // to be implemented
+		break;
+
+	case TFMINI_MODEL::MODEL_TFMINIPLUS: // TFMINI-PLUS (0 to 0xFFFF)
+		if (signal_value < 100 || signal_value == 0xFFFF) {
+			signal_quality = 0;
+			break;
+		}
+
+		//according to the spec, signal is within [0-0xFFFF]
+		//signal_quality = (uint8_t)( (100 * (uint32_t)signal_value) / 0xFFFF);
+		//according to my test, signal is within [0-20000]
+		signal_value = math::min(signal_value, (uint16_t)20000);
+		signal_quality = (int8_t)((100 * (uint32_t)signal_value) / 20000);
```

Comment:
Could you change these c-style casts to static_casts here and elsewhere, e.g.`static_cast<int8_t>(var)` ?